### PR TITLE
Process 'Rehost'

### PR DESCRIPTION
**Pull request by micheljung**
Processes the new `Rehost` command and sends a `game_launch` to the requesting client, as soon as the previous game connection has been closed.

### CODE REVIEW

**Comment by micheljung** [comment]
Successfully tested with local server/client/game

File: server/lobbyconnection.py [hunk 1]
```diff
@@ -937,10 +937,14 @@ def launch_game(self, game, port, is_host=False):
         self.player.game = game
         self.player.game_port = port
 
+        args = ["/numgames " + str(self.player.numGames)]
```
Comment (Sheeo) [inline server/lobbyconnection.py]: This should be client side

**Comment by Sheeo** [comment]
This will invalidate quite a few assumptions in `Game` currently. We need to create a new `Game` object with a fully fresh state to be sure we're not messing up something.

The rehost procedure should do this like the `game_host` function currently does (There's probably some refactoring to be gained there)

File: server/lobbyconnection.py (hunk 1 above)
Comment (micheljung) [inline server/lobbyconnection.py]: The reason I made it server-side is to that it's client-independent. But it's ok for me to have it client side....

**Comment by micheljung** [comment]
Correct me if I'm wrong, but in my PR the procedure IS using game_host, so I don't see your point. It should be exactly like when the client sent a game_host message.

**Comment by Sheeo** [comment]
Right, I missed the part that constructs a game_host message and passes it to lobbyconnection.

I'm against the way this is done -- I'd prefer for the client to intercept the rehost gpg command and issue `game_host` by itself, it could include `rehost: true` in the message so the server can let other clients know that this is a rehosted game. How does that sound?

I want to eventually get rid of the parent-pointer that is gameconnection.lobbyconnection, so this is a step in the wrong direction regarding that.

**Comment by Sheeo** [comment]
We agreed in slack to do this clientside, so I'm closing this now :+1: :)

**Pull request closed without merging**